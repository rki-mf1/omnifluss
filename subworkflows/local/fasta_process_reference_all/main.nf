include { PREPARE_REFERENCE } from '../../../modules/local/inv_consensus_biopython/main'
include { SAMTOOLS_FAIDX    } from '../../../modules/nf-core/samtools/faidx/main'
include { BWA_INDEX         } from '../../../modules/nf-core/bwa/index/main'


workflow FASTA_PROCESS_REFERENCE_ALL {
    take:
    reference_selection           // string
    ch_reads                      // channel: [ val(meta), fastq ]
    ref_path                      // channel: [ val(meta), fasta ]

    main:
    ch_versions = Channel.empty()

    if (reference_selection == "static") {
        ref = tuple([id:ref_path.split("/")[-1].split("\\.")[0]], ref_path) // channel: [ val(meta), fasta ]
    } else if (reference_selection == "mapping") {
        //TODO
    } else {
        {exit 1, "invalid value supplied for variable 'reference_selection' !"}
    }

    PREPARE_REFERENCE(
        ref
    ).preped_ref
    | set {ch_preped_ref}

    ch_versions = ch_versions.mix(PREPARE_REFERENCE.out.versions.first())

    SAMTOOLS_FAIDX(
        ch_preped_ref,
        [[],[]]
    ).fai
    | set {ch_fai_index}

    ch_versions = ch_versions.mix(SAMTOOLS_FAIDX.out.versions.first())

    BWA_INDEX(
        ch_preped_ref
    ).index
    | set {ch_bwa_index}

    ch_versions = ch_versions.mix(BWA_INDEX.out.versions)

    emit:
    preped_ref           = ch_preped_ref
    fai_index            = ch_fai_index
    bwa_index            = ch_bwa_index

    versions             = ch_versions

}

