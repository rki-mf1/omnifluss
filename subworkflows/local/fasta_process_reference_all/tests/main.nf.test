nextflow_workflow {

    name "Test Subworkflow FASTA_PROCESS_REFERENCE_ALL"
    script "../main.nf"
    // Incude a nextflow.config for the test if needed
    config "subworkflows/local/fasta_process_reference_all/tests/nextflow.config"
    workflow "FASTA_PROCESS_REFERENCE_ALL"

    //  Add tags for the subworkflow
    tag "subworkflows"
    tag "subworkflows/local"
    tag "subworkflows/fasta_process_reference_all"
    tag "fasta_process_reference_all"
    // Add tags for all modules used within this subworkflow.
    tag "samtools_faidx"

    // Test name preferably indicating the test-data and file-format used
    test("sarscov2 - single_end - stub") {

        // Activate the stub mode for nextflow
        options '-stub'

        when {
            workflow {
                """
                input[0] = Channel.value([[id:"dummy_file"],"$projectDir/assets/dummy_file.txt"])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match()}
            )
        }
    }

    test("sarscov2 - paired_end - stub") {

        // Activate the stub mode for nextflow
        options '-stub'

        when {
            workflow {
                """
                input[0] = Channel.value([[id:"dummy_file"],"$projectDir/assets/dummy_file.txt"])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match()}
            )
        }
    }
}