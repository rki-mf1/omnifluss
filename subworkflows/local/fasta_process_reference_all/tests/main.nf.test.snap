{
    "sarscov2 - single_end - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "dummy_file"
                        },
                        "dummy_file_preped.fasta:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "dummy_file"
                        },
                        "dummy_file_preped.fasta.fai:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "dummy_file"
                        },
                        [
                            "dummy_file_preped.amb:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "dummy_file_preped.ann:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "dummy_file_preped.bwt:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "dummy_file_preped.pac:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "dummy_file_preped.sa:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "3": [
                    "versions.yml:md5,71a08e942ca2135c3774ed51f3a016f5",
                    "versions.yml:md5,acc04d7444d23c01bfa21ecabe6e054e",
                    "versions.yml:md5,adbc88ab264ac89cad5ade17f4c494de"
                ],
                "bwa_index": [
                    [
                        {
                            "id": "dummy_file"
                        },
                        [
                            "dummy_file_preped.amb:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "dummy_file_preped.ann:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "dummy_file_preped.bwt:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "dummy_file_preped.pac:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "dummy_file_preped.sa:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "fai_index": [
                    [
                        {
                            "id": "dummy_file"
                        },
                        "dummy_file_preped.fasta.fai:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "preped_ref": [
                    [
                        {
                            "id": "dummy_file"
                        },
                        "dummy_file_preped.fasta:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,71a08e942ca2135c3774ed51f3a016f5",
                    "versions.yml:md5,acc04d7444d23c01bfa21ecabe6e054e",
                    "versions.yml:md5,adbc88ab264ac89cad5ade17f4c494de"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.1"
        },
        "timestamp": "2024-08-09T12:08:21.839230824"
    },
    "sarscov2 - paired_end - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "dummy_file"
                        },
                        "dummy_file_preped.fasta:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "dummy_file"
                        },
                        "dummy_file_preped.fasta.fai:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "dummy_file"
                        },
                        [
                            "dummy_file_preped.amb:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "dummy_file_preped.ann:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "dummy_file_preped.bwt:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "dummy_file_preped.pac:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "dummy_file_preped.sa:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "3": [
                    "versions.yml:md5,71a08e942ca2135c3774ed51f3a016f5",
                    "versions.yml:md5,acc04d7444d23c01bfa21ecabe6e054e",
                    "versions.yml:md5,adbc88ab264ac89cad5ade17f4c494de"
                ],
                "bwa_index": [
                    [
                        {
                            "id": "dummy_file"
                        },
                        [
                            "dummy_file_preped.amb:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "dummy_file_preped.ann:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "dummy_file_preped.bwt:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "dummy_file_preped.pac:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "dummy_file_preped.sa:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "fai_index": [
                    [
                        {
                            "id": "dummy_file"
                        },
                        "dummy_file_preped.fasta.fai:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "preped_ref": [
                    [
                        {
                            "id": "dummy_file"
                        },
                        "dummy_file_preped.fasta:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,71a08e942ca2135c3774ed51f3a016f5",
                    "versions.yml:md5,acc04d7444d23c01bfa21ecabe6e054e",
                    "versions.yml:md5,adbc88ab264ac89cad5ade17f4c494de"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.1"
        },
        "timestamp": "2024-08-09T12:08:35.048086423"
    }
}